packages/calypso-e2e/src/lib/components/editor-publish-panel-component.ts

```diff
@@ -0,0 +1,90 @@
+import { Page, Frame, FrameLocator } from 'playwright';
+
+// Options for Playwright's Frame.click() method.
+type ClickOptions = Parameters< Frame[ 'click' ] >[ 1 ];
+type PrePublishButtons = 'Publish' | 'Schedule' | 'Cancel';
+
+const panel = 'div.editor-post-publish-panel';
+const selectors = {
+	// Pre-publish
+	prePublishButton: ( buttonText: PrePublishButtons ) =>
+		`${ panel } .editor-post-publish-panel__header button:text("${ buttonText }")`,
+
+	// Post-publish
+	postPublishClosePanelButton: `${ panel } button[type="button"]:has(svg[aria-hidden="true"])`, // aria-label changes depending on the UI language used.
```

Comment:
Ideally, we develop all components going forward with an eye towards i18n, so that means either:
1. accepting the string to be matched;
2. use selectors that do not rely on UI text;

I've struggled with committing to this, but I plan to review the selectors I've implemented here to see if I can discontinue use of text.

Reply:
I've replaced the whole lot of selectors with non-text based ones.

Reply:
>use selectors that do not rely on UI text;

We might want to spin up a separate thread or P2 to discuss this, but i'm a little hesitant about this as a commitment going forward... This seems to go against our recommendation of using selectors that are user facing. I think our use of text and aria based selectors is one of the things that has made this Playwright transition a lot cleaner than our old Selenium scripts.

What do you think about allowing text-based selecting by default, and then only changing what we need to for I18N compatibility at the time that we need it? I want to avoid lowering the quality of all of our selectors for a potential future use case that may not come.

All that said, I think that the editor is likely something that needs I18N support, and I think what you have here in this PR is just fine, I just wanted to throw that out there for future consideration!

Reply:
Sure, let's spin up a new P2 and involve the I18N folks for some input on this.

In your subsequent paragraphs you did cover my reason for going against the Playwright team's recommendation. Notably, that `aria` attributes and UI text are subject to change depending on the language selected. This would definitely lead to issues when these methods are intended to run against non-US English interfaces, as we've found out already in https://github.com/Automattic/wp-calypso/pull/60584. 

I didn't make the decision lightly, as you say - it does run counter to what we've done so far.

Reply:
Drafted a post and requested a quick look from you before I publish.